ansible/roles/test/files/ptftests/wr_arp.py

```diff
@@ -359,3 +382,121 @@ def req_dut(self, cmd):
+
+            uptime_before = self.get_uptime()
+            self.warm_reboot()
+
+            test_non_broadcast_reply_thread.join(timeout=self.how_long)
+
+            wr_state = self.get_warm_restart_state()
+            if wr_state is not False:
+                self.assertTrue(False, "CPA quit before warm reboot finished")
+
+            if test_non_broadcast_reply_thread.isAlive():
+                self.log("Timed out waiting for test_non_broadcast_reply_thread")
+                self.assertTrue(False, "Timed out waiting for test_non_broadcast_reply_thread")
+
+            uptime_after = self.get_uptime()
+            if uptime_before == uptime_after:
+                self.log("The DUT wasn't rebooted. Uptime: %s vs %s" % (uptime_before, uptime_after))
+                self.assertTrue(uptime_before != uptime_after,
+                                "The DUT wasn't rebooted. Uptime: %s vs %s" % (uptime_before, uptime_after))
+
+    def test_non_broadcast_reply_thr(self, port):
+        pkt, exp_pkt = self.gen_pkts[port]
+
+        # everflow table creation check
+        stop_at = time.time() + self.how_long
+        while True:
+            # 1 second as interval to check if everflow created
+            if self.get_everflow_acl_counter(1) == -1:
+                if time.time() >= stop_at:
+                    self.log("ERR: everflow rule_arp table seems not up")
+                    self.assertTrue(False, "everflow rule_arp table seems not up")
+                else:
+                    continue
+            else:
+                break
+
+        failure_check = 0
+        while True:
+            ef_count_before = self.get_everflow_acl_counter(1)
+            if ef_count_before == -1:
+                if failure_check >= self.RETRIVE_COUNTER_UPLIMIT:
+                    self.log("everflow rule_arp seems closed, test will close")
+                    return
+                else:
+                    failure_check += 1
+                    continue
+
+            # vxlan tunnel should be there
+            self.check_neighbor_advertise_vxlan()
+
+            testutils.send_packet(self, port, pkt)
+            # check arp reply should come from same port
+            # clean arp, test broadcast reply if needed
+            testutils.verify_packet_any_port(self, exp_pkt, ports=[port])
+
+            ef_count_after = self.get_everflow_acl_counter()
+            if ef_count_after == -1:
+                continue
+
+            self.assertTrue(ef_count_after >= ef_count_before + 1,
+                            "Unexpected results, counter_after {} <= counter_before {}".format(
+                                ef_count_after, ef_count_before))
+
+    def get_everflow_acl_counter(self, timeout=COUNTERS_UPDATE_INTERVAL):
+        # Wait for orchagent to update the ACL counters
+        time.sleep(timeout)
+        output, _, _ = self.dut_connection.execCommand('aclshow -a')
+        result = parse_show(output)
```

Comment:
This call will fail, won't it? `parse_show` in your definition accepts two arguments, and only one is passed.

Reply:
Thanks for point out the mistake. Missed when I port the code from mgmt-internal repo to mgmt public.